examples/matrix_multiplication.masm

```diff
@@ -0,0 +1,704 @@
+#
+# Column and row major order matrixes are padded the same way, this procedure
+# copies the data from the advice tape to memory. The value of `words`
+# determines the size of each column/row, and `iterations` how many there are.
+proc.copy_matrix_data_from_tape.0
+    # input: [iterations, words, writePtr, H, H, H, ...]
+    # output: [H, H, H, ...]
+
+    # OUTER: [iterations, words, writePtr, H, H, H, ...]
+
+    # if(iterations>0)
+    dup neq.0 while.true
+
+        # save a copy of `words` for the next iteration, call the loop counter `c`
+        # [c, words, iterations, addr, H, H, H, ...]
+        swap dup
+
+        # INNER: [c, addr, words, iterations, H, H, H, ...]
+        movup.3 swap
+
+        # if(c>0)
+        dup neq.0 while.true
+            # [H, H, H, addr, c, words, iterations, ...]
+            swap movdnw.3
+
+            # [H, H, H, addr, c, words, iterations, ...]
+            adv_pipe
+
+            # INNER: [c, addr, words, iterations, H, H, H, ...]
+            movupw.3 swap
+
+            # every call to `adv_pipe` copies *two* words
+            sub.2
+
+            # while(c>0)
+            dup exec.is_not_zero_or_negative_one
+        end
+
+        # remove the padding word at the end of the row/column
+        eq.18446744069414584320 if.true
+            sub.1
+        end
+
+        # OUTER: [iterations, words, addr, H, H, H, ...]
+        movdn.2 swap
+
+        # while(iterations>0)
+        sub.1 dup neq.0
+    end
+
+    # [H, H, H, ...]
+    drop drop drop
+end
+
+# matrix_multiply(addrA: in Address, addrB: in Address, addrC: out Address) -> void
+#
+# Given 3 memory addresses on the top of the stack, [addrA, addrB, addrC].
+# Where each address is the memory location of a matrices. A, B, C, and
+# matrices A and C are in row-major order, while matrix B is column major.
+# This procedure computes C=A*B.
+#
+# Notes: 
+# - `addrC` must different to `addrA` and `addrB`
+# - This procedure only works with matrixes where both row and col are u32
+# values
+proc.matrix_multiply.0
+    # input: [addrA, addrB, addrC, ...]
+    # output: [...]
+
+    # copy dimensions of A
+    # [[rowA, colA, 0, 0], addrA, addrB, addrC, ...]
+    padw dup.4 mem_loadw
+
+    # copy dimensions of B
+    # [[rowB, colB, 0, 0], [rowA, colA, 0, 0], addrA, addrB, addrC, ...]
+    padw dup.9 mem_loadw
+
+    # assert rowB==colA requirement for matrix multiplication
+    # [[colB, 0, 0, rowA], [colA, 0, 0, addrA], addrB, addrC, ...]
+    dup.5 assert_eq
+
+    # Set C's size, where rowC=rowA colC=colB
+    # [[rowC, colC, 0, 0], [colA, 0, 0, addrA], addrB, addrC, ...]
+    movup.3 dup.9 mem_storew
+
+    # count the number of words the inner most loop have to consume.
+    # [[words, rowC, colC, 0], [0, 0, 0, addrA], addrB, addrC, ...]
+    movup.4 exec.count_to_word
+
+    # discard extra padding in the middle of the data
+    # [[words, rowC, colC, 0], [0, addrA, addrB, addrC], ...]
+    movup.3 drop movup.3 drop
```

Comment:
operations like this made me wish that drop would accept a bitmask to drop multiple items at once. Here it could be `drop.0b00011`

Reply:
Yes, that would be cool! Maybe create a feature-request issue for this so we can track it?

Reply:
Created :) https://github.com/0xPolygonMiden/miden-vm/issues/642